File: app/src/main/java/com/example/khj_pc/gaonnuri/Adapter/ChartRecyclerViewAdapter.java
package com.example.khj_pc.gaonnuri.Adapter;

import android.content.Context;
import android.support.annotation.NonNull;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import com.example.khj_pc.gaonnuri.Data.CircleChart;
import com.example.khj_pc.gaonnuri.R;

import java.util.ArrayList;

public class ChartRecyclerViewAdapter extends RecyclerView.Adapter<ChartRecyclerViewAdapter.ChartRecyclerViewViewHolder> {
    private Context context;
    private ArrayList<CircleChart> datas;
    public  int Position = 0;

    public ChartRecyclerViewAdapter(Context context, ArrayList<CircleChart> datas) {
        this.context = context;
        this.datas = datas;
    }

    @NonNull
    @Override
    public ChartRecyclerViewAdapter.ChartRecyclerViewViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View itemView = LayoutInflater.from(parent.getContext()).inflate(R.layout.item_chart_circle_chart, parent, false);
        return new ChartRecyclerViewAdapter.ChartRecyclerViewViewHolder(itemView);
    }

    @Override
    public void onBindViewHolder(@NonNull ChartRecyclerViewViewHolder holder, int position) {

    }



    @Override
    public int getItemCount() {
        return datas.size();
    }

    public static class ChartRecyclerViewViewHolder extends RecyclerView.ViewHolder {

        View root;


        public ChartRecyclerViewViewHolder(@NonNull View itemView) {
            super(itemView);
            root = itemView;


        }

    }

}

File: app/src/main/java/com/example/khj_pc/gaonnuri/Adapter/ChartRecyclerViewAdapter.kt
package com.example.khj_pc.gaonnuri.Adapter

import android.content.Context
import android.support.v7.widget.RecyclerView
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup

import com.example.khj_pc.gaonnuri.Data.CircleChart
import com.example.khj_pc.gaonnuri.R
import kotlinx.android.synthetic.main.item_chart_circle_chart.view.*

class ChartRecyclerViewAdapter(private val context: Context, private val datas: ArrayList<CircleChart>) : RecyclerView.Adapter<ChartRecyclerViewAdapter.ChartRecyclerViewViewHolder>() {
    var Position = 0

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ChartRecyclerViewAdapter.ChartRecyclerViewViewHolder {
        val itemView = LayoutInflater.from(parent.context).inflate(R.layout.item_chart_circle_chart, parent, false)
        return ChartRecyclerViewAdapter.ChartRecyclerViewViewHolder(itemView)
    }

    override fun onBindViewHolder(holder: ChartRecyclerViewViewHolder, position: Int) {
        val data = datas[position]
        holder.itemView.title.setText(data.title)
        holder.itemView.content.setText(data.content)
        holder.itemView.donut_progress.setDonut_progress(data.percent.toString())
    }


    override fun getItemCount(): Int {
        return datas.size
    }

    class ChartRecyclerViewViewHolder(internal var root: View) : RecyclerView.ViewHolder(root)

}
